Add a 3D tonemapping and color correction demo

Calinou opened the pull request:
This showcases various tonemapping operators and their interaction with several color correction textures (1D and 3D).

A script generating neutral 3D LUT textures is also included.

## Preview

![tonemap_color_correction](https://user-images.githubusercontent.com/180032/221329438-a06d3481-bdf0-424f-b27f-e1858f246069.png)

huwpascoe commented on 3d/tonemap_color_correction/neutral_luts/3d/create_neutral_3d_luts.gd:
```diff
@@ -0,0 +1,47 @@
+## Lower resolution neutral LUTs such as 17×17×17 and 33×33×33 will still somewhat impact rendering
+## due to the limitations of bilinear filtering.
+func create_neutral_lut(p_name: String, p_size: int, p_vertical: bool):
+	var image = Image.create(
```
var image = Image.create_empty(

huwpascoe commented on 3d/tonemap_color_correction/options.gd:
```diff
@@ -0,0 +1,72 @@
+
+
+func _on_tonemap_mode_item_selected(index: int) -> void:
+	world_environment.environment.tonemap_mode = index
```
world_environment.environment.tonemap_mode = index as Environment.ToneMapper

huwpascoe reviewed (changes requested):
This is a very nice demo. Only needs two changes for latest Godot.

Calinou commented:
This PR now integrates https://github.com/Calinou/godot-demo-projects/pull/4 courtesy of @allenwp 🙂

Calinou commented:
I've updated the PR to add AgX support.

AThousandShips commented on 3d/tonemap_color_correction/options.gd:
```diff
@@ -0,0 +1,72 @@
+	# Hide whitepoint if not relevant (Linear tonemapping does not use a whitepoint).
+	%Whitepoint.visible = world_environment.environment.tonemap_mode != Environment.TONE_MAPPER_LINEAR and world_environment.environment.tonemap_mode != Environment.TONE_MAPPER_AGX
+
+func _on_exposure_value_changed(value: float) -> void:
```
```suggestion

func _on_exposure_value_changed(value: float) -> void:
```

allenwp commented:
> I've updated the PR to add AgX support following the merge of [godotengine/godot#87260](https://github.com/godotengine/godot/pull/87260).

I think you accidentally reverted https://github.com/Calinou/godot-demo-projects/pull/4 ?

Calinou commented:
> I think you accidentally reverted [Calinou#4](https://github.com/Calinou/godot-demo-projects/pull/4) ?

Fixed.

Calinou commented:
Rebased and tested again on 4.4.1.stable, it works as expected. I've also added a project icon as it was previously missing.

Merging, since this was recently approved. Once https://github.com/godotengine/godot/pull/94496 is ready to merge, we can modify the demo to have HDR support and add more test scenes to better showcase HDR functionality.
